### ISSUE TITLE
Crash when deleting a bar inside Musescore Flatpak on Fedora

### ISSUE THREAD

**Issue by Sytoplis** [musescore/MuseScore#25619]
### Issue type

Crash or freeze

### Description with steps to reproduce

1. Install Musescore Flatpak (MuseScore Version 4.4.2)
[FlatpakSettings.zip](https://github.com/user-attachments/files/17879494/FlatpakSettings.zip)
2. Have a 3 Instrument Score
3. Place Notes in a particular Bar (I havent really figured out what bar exactly, but there is a link to a File with which I was able to reliably reproduce this bug)
[DeleteCrash.zip](https://github.com/user-attachments/files/17879479/DeleteCrash.zip)
[DeleteCrash.pdf](https://github.com/user-attachments/files/17879502/DeleteCrash.pdf)
4. Delete some bar (in my example its the last bar with notes [Should be measure 15] you delete to crash MuseScore)

### Supporting files, videos and screenshots

[MuseScore_241123_124920.log](https://github.com/user-attachments/files/17879544/MuseScore_241123_124920.log)
[MuseScoreDiagnostics.zip](https://github.com/user-attachments/files/17879560/MuseScoreDiagnostics.zip)

### What is the latest version of MuseScore Studio where this issue is present?

4.4.2

### Regression

I was unable to check

### Operating system

Fedora 40

### Additional context

I am sorry that i am reporting a bug from a Flatpak. I know this isn't supported here (but its honestly the easiest installation on Linux, but that's another topic), but I thought this problem might be reproducible on other systems as well.

I also have the beta of the Muse Sounds Manager installed, but i don't know if that effects the Bug at all, as I have not used it to install any Extensions (I don't have any Extensions).
Sadly i have my Fedora System freshly set up and don't have a tool yet to do screen recordings with (There are some problems with Wayland).

Further Information on the Bug itself:
- It doesn't crash with 2 Instruments, but I can also get it to crash with 4
- From the logs i would expect that the layout computation causes the crash, as something with the width is wrong, but what do i know.

Is probably related to:
- https://github.com/musescore/MuseScore/issues/24705
- https://github.com/musescore/MuseScore/issues/24648
But i opened another issue, as i don't know for sure, if its the same bug.

Could be related to:
- https://github.com/musescore/MuseScore/issues/22711

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by cbjeukendrup** [comment]
I can't reproduce it on Mac. It would be useful to get a stack trace of the crash. Also, when you run from the command line, do you get any other output than what's in the logs? For example, at the moment that it crashes, does it say something about an assertion failure?

**Comment by MarcSabatella** [comment]
No crash on Linux App image either.  But the situation seems identical to https://github.com/musescore/MuseScore/issues/24705 - an operation that causes the last system to become empty.

FYI, installing the App image is really extremely simple:

1) download
2) chmod +x
3) run with "install" option

**Comment by Sytoplis** [comment]
Ok, that seems to be a Flatpak issue then? Ill probably switch to the AppImage after this, but ill help you figure this out first. So I ran musescore now via the Cosole and it seems to be exactly like #24705, i get:
`/usr/include/c++/14.2.0/bits/stl_vector.h:1237: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::back() [with _Tp = mu::engraving::MeasureBase*; _Alloc = std::allocator<mu::engraving::MeasureBase*>; reference = mu::engraving::MeasureBase*&]: Assertion '!this->empty()' failed.`
I dont know where to find the stack trace tho. I didnt get any other output or popup except the logs when i just launched it via rofi.

**Comment by Sytoplis** [comment]
So i got my debugger going an here's the backtrace:
```
#0  0x00007ffff489ad94 in __pthread_kill_implementation () at /usr/lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffff484208e in raise () at /usr/lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffff4829882 in abort () at /usr/lib/x86_64-linux-gnu/libc.so.6
#3  0x00007ffff4ae8c1d in std::__glibcxx_assert_fail(char const*, int, char const*, char const*) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x000055555671dfa5 in ??? ()
#5  0x00005555566a1068 in ??? ()
#6  0x00005555566a25d8 in ??? ()
#7  0x00005555566a020e in ??? ()
#8  0x00005555563bee5a in ??? ()
#9  0x00005555561e9cc6 in ??? ()
#10 0x00005555561ecea0 in ??? ()
#11 0x0000555556a214b4 in ??? ()
#12 0x0000555556c391bd in ??? ()
#13 0x0000555556c3c29d in ??? ()
#14 0x0000555556c1e73a in ??? ()
#15 0x000055555689fb4f in ??? ()
#16 0x00005555568c8e42 in ??? ()
#17 0x00005555568c0c2e in ??? ()
#18 0x00005555568d214f in ??? ()
#19 0x00007ffff6ff6bca in ??? () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#20 0x00007ffff6e86957 in QV4::QObjectMethod::callPrecise(QQmlObjectOrGadget const&, QQmlPropertyData const&, QV4::ExecutionEngine*, QV4::CallData*, QMetaObject::Call) () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#21 0x00007ffff6e887a1 in QV4::QObjectMethod::callInternal(QV4::Value const*, QV4::Value const*, int) const () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#22 0x00007ffff6ef9f78 in ??? () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#23 0x00007ffff6eff51f in ??? () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#24 0x00007ffff6e24d24 in ??? () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#25 0x00007ffff6e25859 in QV4::Function::call(QObject*, void**, QMetaType const*, int, QV4::ExecutionContext*) () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#26 0x00007ffff6fad717 in QQmlJavaScriptExpression::evaluate(void**, QMetaType const*, int) () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#27 0x00007ffff6f45173 in QQmlBoundSignalExpression::evaluate(void**) () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#28 0x00007ffff6f45ba0 in ??? () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#29 0x00007ffff6fde1d5 in QQmlNotifier::emitNotify(QQmlNotifierEndpoint*, void**) () at /usr/lib/x86_64-linux-gnu/libQt6Qml.so.6
#30 0x00007ffff50052f8 in ??? () at /usr/lib/x86_64-linux-gnu/libQt6Core.so.6
#31 0x00007ffff78e9c48 in QQuickShortcut::event(QEvent*) () at /usr/lib/x86_64-linux-gnu/libQt6Quick.so.6
#32 0x00007ffff61a0238 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/x86_64-linux-gnu/libQt6Widgets.so.6
#33 0x00007ffff4f9d8d8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/x86_64-linux-gnu/libQt6Core.so.6
#34 0x00007ffff5c9ec48 in QShortcutMap::dispatchEvent(QKeyEvent*) () at /usr/lib/x86_64-linux-gnu/libQt6Gui.so.6
#35 0x00007ffff5c9fb6f in QShortcutMap::tryShortcut(QKeyEvent*) () at /usr/lib/x86_64-linux-gnu/libQt6Gui.so.6
#36 0x00007ffff58654f8 in QWindowSystemInterface::handleShortcutEvent(QWindow*, unsigned long, int, QFlags<Qt::KeyboardModifier>, unsigned int, unsigned int, unsigned int, QString const&, bool, unsigned short) () at /usr/lib/x86_64-linux-gnu/libQt6Gui.so.6
#37 0x00007ffff57f49ce in QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) () at /usr/lib/x86_64-linux-gnu/libQt6Gui.so.6
#38 0x00007ffff586051c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/x86_64-linux-gnu/libQt6Gui.so.6
#39 0x00007ffff10924e6 in ??? () at /usr/lib/x86_64-linux-gnu/libQt6XcbQpa.so.6
#40 0x00007ffff47143f6 in g_main_dispatch.lto_priv () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#41 0x00007ffff4774f87 in g_main_context_iterate_unlocked.isra () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#42 0x00007ffff4713883 in g_main_context_iteration () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#43 0x00007ffff52aaae3 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/x86_64-linux-gnu/libQt6Core.so.6
#44 0x00007ffff4fac0db in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/x86_64-linux-gnu/libQt6Core.so.6
#45 0x00007ffff4fa78ce in QCoreApplication::exec() () at /usr/lib/x86_64-linux-gnu/libQt6Core.so.6
#46 0x0000555555b6067f in ??? ()
#47 0x00007ffff482b188 in __libc_start_call_main () at /usr/lib/x86_64-linux-gnu/libc.so.6
#48 0x00007ffff482b24b in __libc_start_main () at /usr/lib/x86_64-linux-gnu/libc.so.6
#49 0x00005555560380b5 in ??? ()
```

**Comment by cbjeukendrup** [comment]
Thanks for checking! Given the assertion failure message, and considering the reproduction steps, it seems highly probable that this is indeed the same as https://github.com/musescore/MuseScore/issues/24648. I'll close this as a duplicate. We expect to have a fix in version 4.5 or the release after that. For now, the workaround would be to stick to the AppImage.
